scripts/cube_fx.lua

```diff
@@ -290,11 +300,59 @@ local function cube_victory(size, results)
   end
 end
 
+-- Map to determine if some machine is using the cube
+-- in an "efficient" manner. 
+local cube_machine_condition_handlers = {
+  ["status"] = function(entity, cube_type)
+    return entity.status == defines.entity_status.working
+  end,
+  ["handcrafting"] = function(character, cube_type)
+    -- We could search through the crafting queue, but this function
+    -- needs to be quick because it's called every 6 ticks. Lets
+    -- rather assume that if the player doesn't have it in their
+    -- inventory or cursor then the player is crafting with it.
+    -- That doesn't handle it being in queues though, but it's
+    -- better than nothing.
+    local player = character.player
+    if not player then return end
+    local cursor_stack = player.cursor_stack
+    if cursor_stack.valid and cursor_stack.valid_for_read then
+      if cursor_stack.name == cube_type then return false end
+    end
+    local inventory = player.get_main_inventory()
+    if inventory.find_item_stack(cube_type) then return false end
+    return true
+  end,
+}
+
+-- Map to determine in which machine types the cube can be
+-- seen as being "utilization". The value will show what 
+-- requirement this machine has to be "productive" with
+-- the cube. 
+local cube_working_machine_types = {
+  ["character"]           = cube_machine_condition_handlers["handcrafting"],
+  ["assembling-machine"]  = cube_machine_condition_handlers["status"],
+  ["furnace"]             = cube_machine_condition_handlers["status"],
+  ["boiler"]              = cube_machine_condition_handlers["status"],
+  ["generator"]           = cube_machine_condition_handlers["status"],
+  ["burner-generator"]    = cube_machine_condition_handlers["status"],
+  ["rocket-silo"]         = cube_machine_condition_handlers["status"],
+  ["reactor"]             = cube_machine_condition_handlers["status"], -- Does this work?
+}
+
 local function track_victory_statistics(size, results)
-  if size ~= 1 then return end  -- Only track distance travelled by main cube.
+  if size == 0 then return end      -- There are no cubes! That's weird, let's ignore the problem
+
+  -- We will only look at what the first cube is doing, because there might be
+  -- multiple like when it's in its phantom form. This makes it not entirely
+  -- accurate, but it's better than nothing. It's also important that we track
+  -- something as often as possible to ensure the "utilization" is calculated
+  -- somewhat correctly.
   local entity = results[1].entity
+
   if not entity or not entity.valid then return end
 
+  -- Track distance travelled by this cube
```

Comment:
I think we cannot attempt to track the distance travelled when there are multiple cubes since the ordering of results is not guaranteed to be consistent. The first result may be the position of some cube one tick and the position of a totally different cube the next. I think we should just clear the `cube_last_position` whenever there is more than one.

Reply:
Yeah, you're right. Good catch. Will stop tracking when the cube is split apart.